packages/public-api/src/packageInfo/packageSummaryService.ts

```diff
@@ -0,0 +1,73 @@
+import { getRepository } from 'typeorm';
+import {
+  PackageMetadata,
+  PackageUsageByHostnameProjection,
+  PackagePopularityView,
+  PackageVulnerability,
+  toSerializable,
+} from '@gradejs-public/shared';
+import semver from 'semver';
+
+export async function getPackageSummaryByName(packageName: string) {
+  const packageRepo = getRepository(PackageMetadata);
+  const packageUsageRepo = getRepository(PackageUsageByHostnameProjection);
+  const packagePopularityRepo = getRepository(PackagePopularityView);
+  const packageVulnerabilities = getRepository(PackageVulnerability);
+
+  const queries = [
+    packageRepo.createQueryBuilder().where('name = :packageName', { packageName }),
+    packageUsageRepo
+      .createQueryBuilder('usage')
+      .leftJoinAndSelect('usage.hostname', 'hostname')
+      .where('package_name = :packageName', { packageName })
+      .limit(16), // TODO: remove hardcode
+    packagePopularityRepo
+      .createQueryBuilder()
+      .where('package_name = :packageName', { packageName }),
+    packageVulnerabilities
+      .createQueryBuilder()
+      .where('package_name = :packageName', { packageName }),
+  ];
+
+  const [[packageInfo], usage, [popularity], vulnerabilities] = (await Promise.all(
```

Comment:
This line looks quite weird, especially the destructuring assignment.
Can we do something like:
```
const queries = [
   (...).getMany() // or getRawMany();
]
const [packageInfo, usage, ...] = await Promise.all(queries);
```